parsing/lexer.mll

```diff
@@ -300,6 +301,10 @@ let prepare_error loc = function
       Location.errorf ~loc
         "This comment contains an unterminated string literal"
         ~sub:[Location.msg ~loc:literal_loc "String literal begins here"]
+  | Empty_character_literal ->
+      Location.errorf ~loc
+        "Illegal empty character literal ''\n\
+         Hint: Did you mean ' ' or a type variable 'a?"
```

Comment:
Nitpicking: the hint could be a suberror message:
```ocaml
Location.error ~loc ~sub:[Location.msg "Hint: Did you mean ' ' or a type variable 'a?"]
  "Illegal empty character literal ''
```

Reply:
Done.